### PR TITLE
Remove JedisPoolAbstract class, hide Pool.initPool() method and related changes

### LINKED ISSUE #1799: JedisSentinelPool function initPool not thread safe
### Expected behavior
currentHostMaster and factory lead to same master address.

### Actual behavior

MasterListener thread will call function initPool when master change.
If mult masterListeners call initPool at the same time with difference params,the currentHostMaster and factory lead to difference master.
So the function getResource will never return.

### Steps to reproduce:

Please create a reproducible case of your problem. Make sure
that case repeats consistently and it's not random
1.
2.
3.

#### Jedis version:
2.7.2

### PR DESCRIPTION
**Pull request by sazzad16**
JedisPoolAbstract itself was doing nothing except acting as access point of Pool\<Jedis\> from redis.clients.jedis package. Using Pool\<Jedis\> directly seems better.

Resolves #1799

### CODE REVIEW

**Comment by sazzad16** [comment]
@HeartSaVioR @marcosnils ?

**Review by bsnyder788 — DISMISSED** [review]

**Comment by bsnyder788** [comment]
@sazzad16 there are some merge conflicts in here though

**Comment by sazzad16** [comment]
@marcosnils @gkorland ping

**Review by dengliming — DISMISSED** [review]

**Review by yangbodong22011 — DISMISSED** [review]

File: src/main/java/redis/clients/jedis/JedisPool.java [hunk 1]
```diff
@@ -29,43 +30,36 @@ public JedisPool(String host, int port) {
...
+        Protocol.DEFAULT_TIMEOUT, null));
   }
 
   /**
+   * WARNING: This constructor only accepts a uri string as {@code url}.
+   * {@link JedisURIHelper#isValid(java.net.URI)} can be used before this.
+   *
+   * To use a host string,
+   * {@link #JedisPool(java.net.URI, javax.net.ssl.SSLSocketFactory, javax.net.ssl.SSLParameters, javax.net.ssl.HostnameVerifier)}
+   * can be used with {@link Protocol#DEFAULT_PORT}.
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisPool.java:52]:
```suggestion
   * {@link #JedisPool(java.lang.String, int, boolean, javax.net.ssl.SSLSocketFactory, javax.net.ssl.SSLParameters, javax.net.ssl.HostnameVerifier)}
   * can be used with {@link Protocol#DEFAULT_PORT} and {@code ssl=true}.
```

File: src/main/java/redis/clients/jedis/JedisPool.java [hunk 2]
```diff
@@ -29,43 +30,36 @@ public JedisPool(String host, int port) {
...
-          Protocol.DEFAULT_TIMEOUT, Protocol.DEFAULT_TIMEOUT, null, Protocol.DEFAULT_DATABASE, null));
-    }
+    this(new GenericObjectPoolConfig<Jedis>(), new JedisFactory(URI.create(url), Protocol.DEFAULT_TIMEOUT,
+        Protocol.DEFAULT_TIMEOUT, null));
   }
 
   /**
+   * WARNING: This constructor only accepts a uri string as {@code url}.
+   * {@link JedisURIHelper#isValid(java.net.URI)} can be used before this.
+   *
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisPool.java:49]:
```suggestion
   * <p>
```

File: src/main/java/redis/clients/jedis/JedisPool.java [hunk 3]
```diff
@@ -29,43 +30,36 @@ public JedisPool(String host, int port) {
...
```
Comment (sazzad16) [inline src/main/java/redis/clients/jedis/JedisPool.java:35]:
```suggestion
   * <p>
```

**Pull request merged**